plugins/modules/ndo_tenant_custom_qos_policy.py

```diff
@@ -0,0 +1,462 @@
+        aliases: [ to ]
+      dscp_target:
+        description:
+        - The DSCP target value.
+        type: str
+        choices:
+          - af11
+          - af12
+          - af13
+          - af21
+          - af22
+          - af23
+          - af31
+          - af32
+          - af33
+          - af41
+          - af42
+          - af43
+          - cs0
+          - cs1
+          - cs2
+          - cs3
+          - cs4
+          - cs5
+          - cs6
+          - cs7
+          - expedited_forwarding
+          - unspecified
+          - voice_admit
+        aliases: [ target ]
+      target_cos:
+        description:
+        - The target CoS to be driven based on the range of input values of Dot1P coming into the fabric.
+        type: str
+        choices: [ background, best_effort, excellent_effort, critical_applications, video, voice, internetwork_control, network_control, unspecified ]
+      qos_priority:
+        description:
+        - The desired QoS class level to be used.
+        type: str
+        choices: [ level1, level2, level3, level4, level5, level6, unspecified ]
+        aliases: [ priority, prio ]
+  state:
+    description:
+    - Use C(absent) for removing.
+    - Use C(query) for listing an object or multiple objects.
+    - Use C(present) for creating or updating.
+    type: str
+    choices: [ absent, query, present ]
+    default: query
+notes:
+- The O(template) must exist before using this module in your playbook.
+  Use M(cisco.mso.ndo_template) to create the Tenant template.
+seealso:
+- module: cisco.mso.ndo_template
+extends_documentation_fragment: cisco.mso.modules
+"""
+
+EXAMPLES = r"""
+- name: Create a new Custom QoS Policy object
+  cisco.mso.ndo_tenant_custom_qos_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: tenant_template
+    name: custom_qos_policy_1
+    dscp_mappings:
+      - dscp_from: af11
+        dscp_to: af12
+        dscp_target: af11
+        target_cos: background
+        qos_priority: level1
+    cos_mappings:
+      - dot1p_from: background
+        dot1p_to: best_effort
+        target: af11
+        target_cos: background
+        qos_priority: level1
+    state: present
+  register: custom_qos_policy_1
+
+- name: Update a Custom QoS Policy object name with UUID
+  cisco.mso.ndo_tenant_custom_qos_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: tenant_template
+    name: custom_qos_policy_2
+    uuid: "{{ custom_qos_policy_1.current.uuid }}"
+    state: present
+
+- name: Query a Custom QoS Policy object with name
+  cisco.mso.ndo_tenant_custom_qos_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: tenant_template
+    name: custom_qos_policy_1
+    state: query
+  register: query_uuid
```

Comment:
shoudl we use query_name here since we do not query with uuid?
```suggestion
  register: query_name
```

Reply:
changed